setup.py

```diff
@@ -26,6 +26,11 @@ def get_version():
     long_description=open('README.rst').read(),
     url='https://github.com/scrapinghub/scrapy-autoextract',
     packages=find_packages(),
+    install_requires=[
+        'autoextract-poet>=0.0.1',
```

Comment:
I wonder if these should be installed by default, even if the user does not want it.
It can help to reduce the end package size, on the user side.
Also, we might want to use `scrapy-autoextract` in a python version that doesn't support asyncio, so not sure if it can be a problem

Reply:
> I wonder if these should be installed by default, even if the user does not want it.

You mean, if users are going to use the middleware approach, this library will be installed as if they were going to use providers, right?

Well, I'm suspect and probably biased, but I believe libraries should include all its dependencies even if users end up not using all of them. It represents less trouble and makes it easier for users when they decide it's time to try the other way around. 

The alternative would be something like we have/used to have on Spidermon, something like `pip install spidermon[monitoring]` but I'm not a big fan of this approach.

> It can help to reduce the end package size, on the user side.

I think the difference in package size would be negligible.

> Also, we might want to use scrapy-autoextract in a python version that doesn't support asyncio, so not sure if it can be a problem

I'm not sure it would be a good idea to use those Scrapy providers in a sync way because it could block our requests cycle.

Reply:
> I'm not sure it would be a good idea to use those Scrapy providers in a sync way because it could block our requests cycle. In any case, I believe asyncio is supported at least since Python 3.6, and Python 3.5 is currently deprecated.

Yeah, but what if I just want to use the middleware version?
And I'm in version 3.4 and a new version of this library is released with a fix in the middleware?

> I think the difference in package size would be negligible.

No hard feelings on this one.
Just that we count the popularity of a package by downloads :stuck_out_tongue: 

About the installation, it will depend on the first case, if we want to install async things by default, even if it may break future fixes..